events/janus_gelfevh.c

```diff
@@ -0,0 +1,606 @@
+	struct addrinfo *res = NULL;
+	janus_network_address addr;
+	janus_network_address_string_buffer addr_buf;
+	struct sockaddr_in servaddr;
+
+	if(getaddrinfo(backend, NULL, NULL, &res) != 0 ||
+				janus_network_address_from_sockaddr(res->ai_addr, &addr) != 0 ||
+				janus_network_address_to_string_buffer(&addr, &addr_buf) != 0) {
+		if(res)
+			freeaddrinfo(res);
+		JANUS_LOG(LOG_ERR, "Could not resolve address (%s)...\n", backend);
+		return -1;
+	}
+	const char *host = g_strdup(janus_network_address_string_from_buffer(&addr_buf));
+	freeaddrinfo(res);
+
+	if((sockfd = socket(AF_INET, transport, 0)) < 0 ) {
+		JANUS_LOG(LOG_ERR, "Socket creation failed: %s\n", strerror(errno));
+		return -1;
+	}
+
+	memset(&servaddr, 0, sizeof(servaddr));
+
+	servaddr.sin_family = AF_INET;
+	servaddr.sin_port = htons(atoi(port));
+	servaddr.sin_addr.s_addr = inet_addr(host);
+
+	if(connect(sockfd, (struct sockaddr *)&servaddr, sizeof(servaddr)) < 0) {
+		JANUS_LOG(LOG_ERR, "Connect to GELF host failed\n");
+		return -1;
+	}
+	JANUS_LOG(LOG_INFO, "Connected to GELF backend: [%s:%s]\n", host, port);
+	return 1;
+}
+
+static int janus_gelfevh_send(char *message) {
+	if(!message) {
+		JANUS_LOG(LOG_WARN, "Message is NULL, not sending to GELF!\n");
+		return -1;
+	}
+	if(transport == JANUS_GELFEVH_SOCKET_TYPE_TCP) {
+		/* TCP */
+		int out_bytes = 0;
+		int length = strlen(message);
+		char *buffer = message;
+		while(length > 0) {
+			out_bytes = send(sockfd, buffer, length + 1, 0);
+			if(out_bytes <= 0) {
+				JANUS_LOG(LOG_WARN, "Sending TCP message failed, dropping event: %s \n", strerror(errno));
+				close(sockfd);
+				return -1;
+			}
+			buffer += out_bytes;
+			length -= out_bytes;
+		}
+	} else {
+		/* UDP chunking with headers. Check if we need to compress the data */
+		int len = strlen(message);
+		char *buf = message;
+		if(compress) {
+			char compressed_text[8192];
+			size_t compressed_len = 0;
+			compressed_len = janus_gzip_compress(compression,
+				message, strlen(message),
+				compressed_text, sizeof(compressed_text));
+			if(compressed_len == 0) {
+				JANUS_LOG(LOG_WARN, "Failed to compress event (%zu bytes). Sending message uncompressed\n", strlen(message));
+				/* Sending message uncompressed */
+			} else {
+				len = compressed_len;
+				buf = compressed_text;
+			}
+		}
+
+		int total = len / max_gelf_msg_len + 1;
+		if(total > MAX_GELF_CHUNKS) {
+			JANUS_LOG(LOG_WARN, "Event not sent! GELF allows %d number of chunks, try increasing max_gelf_msg_len\n", MAX_GELF_CHUNKS);
+			return -1;
+		}
+		/* Do we need to chunk the message */
+		if(total == 1) {
+			int n = send(sockfd, buf, len, 0);
+			if(n < 0) {
+				JANUS_LOG(LOG_WARN, "Sending UDP message failed, dropping event: %s \n", strerror(errno));
+				return -1;
+			}
+			return 1;
+		} else {
+			int offset = 0;
+			char *rnd = randstring(8);
+			for (int i = 0; i < total; i++) {
+				int bytesToSend = offset + max_gelf_msg_len < len ? max_gelf_msg_len : len - offset;
```

Comment:
I think it's better to wrap the items with a math operation in brackets, here, or there may be ambiguities: if not for the compiler, for who reads the code for sure.